tool/tsh/putty_config_windows.go

```diff
@@ -0,0 +1,384 @@
+			return trace.BadParameter("Cannot find registered leaf cluster %q. Use the leaf cluster name as it appears in the output of `tsh clusters`.", cf.LeafClusterName)
+		}
+		certificateFilePath = keypaths.SSHCertPath(keysDir, proxyHost, tc.Config.Username, cf.LeafClusterName)
+	}
+
+	// get all CAs for the cluster (including trusted clusters)
+	var cas []types.CertAuthority
+	err = tc.WithRootClusterClient(cf.Context, func(clt auth.ClientI) error {
+		cas, err = clt.GetCertAuthorities(cf.Context, types.HostCA, false /* exportSecrets */)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		return nil
+	})
+	// iterate over CAs
+	hostCAPublicKeys := make(map[string][]string)
+	for _, ca := range cas {
+		var publicKeys []string
+		// if this is either the root or requested leaf cluster, process it
+		if ca.GetName() == rootClusterName || ca.GetName() == cf.LeafClusterName {
+			for i, key := range ca.GetTrustedSSHKeyPairs() {
+				log.Debugf("%v CA [%v]: %v", ca.GetName(), i, key)
+				kh, err := sshutils.MarshalKnownHost(sshutils.KnownHost{
+					Hostname:      ca.GetClusterName(),
+					AuthorizedKey: key.PublicKey,
+				})
+				if err != nil {
+					return trace.Wrap(err)
+				}
+				_, _, hostCABytes, _, _, err := ssh.ParseKnownHosts([]byte(kh))
+				if err != nil {
+					return trace.Wrap(err)
+				}
+				publicKeys = append(publicKeys, strings.TrimPrefix(strings.TrimSpace(string(ssh.MarshalAuthorizedKey(hostCABytes))), constants.SSHRSAType+" "))
+				hostCAPublicKeys[ca.GetName()] = publicKeys
+			}
+		}
+	}
+
+	hostname := tc.Config.Host
+	port := tc.Config.HostPort
+	userHostString := hostname
+	login := ""
+	if tc.Config.HostLogin != "" {
+		login = tc.Config.HostLogin
+		userHostString = fmt.Sprintf("%v@%v", login, userHostString)
+	}
+
+	// add all host CA public keys for cluster
+	for cluster, hostCAs := range hostCAPublicKeys {
+		keyName := fmt.Sprintf(`TeleportHostCA-%v`, cluster)
+		for i, publicKey := range hostCAs {
+			// append indices to entries if we have multiple public keys for a CA
+			if len(hostCAs) > 1 {
+				keyName = fmt.Sprintf(`%v-%d`, keyName, i)
+			}
+
+			if ok, err := addHostCAPublicKey(keyName, publicKey, hostname); !ok {
+				log.Errorf("Failed to add host CA key for %v: %T", cluster, err)
+				return trace.Wrap(err)
+			} else {
+				log.Debugf("Added/updated host CA key %d for %v", i, cluster)
+			}
+		}
+	}
+
+	proxyCommandClusterName := rootClusterName
+	if cf.LeafClusterName != "" {
+		proxyCommandClusterName = cf.LeafClusterName
+	}
+
+	// format local command string (to run 'tsh proxy ssh')
+	localCommandString := formatLocalCommandString(cf.executablePath, proxyCommandClusterName)
+
+	// add session to registry
+	if ok, err := addPuTTYSession(proxyHost, tc.Config.Host, port, login, ppkFilePath, certificateFilePath, localCommandString, cf.LeafClusterName); !ok {
+		log.Errorf("Failed to add PuTTY session for %v: %T\n", userHostString, err)
+		return trace.Wrap(err)
+	} else {
```

Comment:
Some general advice - you should feel bad every time you have to use `else`, and think of ways to eliminate it.

In this case, it's trivial - the if branch always returns, so you can just remove the else block and "unindent" the code it contains.